homeassistant/components/light/lifx.py

```diff
@@ -6,124 +6,97 @@
+REQUIREMENTS = ['aiolifx==0.4.1.post1']
 
-BYTE_MAX = 255
+UDP_BROADCAST_PORT = 56700
+
+# Delay (in ms) expected for changes to take effect in the physical bulb
+BULB_LATENCY = 500
 
-CONF_BROADCAST = 'broadcast'
 CONF_SERVER = 'server'
 
+BYTE_MAX = 255
 SHORT_MAX = 65535
 
-TEMP_MAX = 9000
-TEMP_MAX_HASS = 500
-TEMP_MIN = 2500
-TEMP_MIN_HASS = 154
-
 SUPPORT_LIFX = (SUPPORT_BRIGHTNESS | SUPPORT_COLOR_TEMP | SUPPORT_RGB_COLOR |
                 SUPPORT_TRANSITION)
 
 PLATFORM_SCHEMA = PLATFORM_SCHEMA.extend({
-    vol.Optional(CONF_SERVER, default=None): cv.string,
-    vol.Optional(CONF_BROADCAST, default=None): cv.string,
+    vol.Optional(CONF_SERVER, default='0.0.0.0'): cv.string,
 })
 
 
 # pylint: disable=unused-argument
-def setup_platform(hass, config, add_devices, discovery_info=None):
+@asyncio.coroutine
+def async_setup_platform(hass, config, async_add_devices, discovery_info=None):
     """Setup the LIFX platform."""
+    import aiolifx
+
     server_addr = config.get(CONF_SERVER)
-    broadcast_addr = config.get(CONF_BROADCAST)
 
-    lifx_library = LIFX(add_devices, server_addr, broadcast_addr)
+    lifx_library = LIFX(hass, async_add_devices)
 
-    # Register our poll service
-    track_time_change(hass, lifx_library.poll, second=[10, 40])
+    coro = hass.loop.create_datagram_endpoint(
+        partial(aiolifx.LifxDiscovery, hass.loop, lifx_library),
+        local_addr=(server_addr, UDP_BROADCAST_PORT))
 
-    lifx_library.probe()
+    hass.loop.create_task(coro)
+    return True
 
 
 class LIFX(object):
-    """Representation of a LIFX light."""
-
-    def __init__(self, add_devices_callback, server_addr=None,
-                 broadcast_addr=None):
-        """Initialize the light."""
-        import liffylights
-
-        self._devices = []
-
-        self._add_devices_callback = add_devices_callback
-
-        self._liffylights = liffylights.LiffyLights(
-            self.on_device, self.on_power, self.on_color, server_addr,
-            broadcast_addr)
-
-    def find_bulb(self, ipaddr):
-        """Search for bulbs."""
-        bulb = None
-        for device in self._devices:
-            if device.ipaddr == ipaddr:
-                bulb = device
-                break
-        return bulb
+    """Representation of all known LIFX entities."""
 
-    def on_device(self, ipaddr, name, power, hue, sat, bri, kel):
+    def __init__(self, hass, async_add_devices):
         """Initialize the light."""
-        bulb = self.find_bulb(ipaddr)
-
-        if bulb is None:
-            _LOGGER.debug("new bulb %s %s %d %d %d %d %d",
-                          ipaddr, name, power, hue, sat, bri, kel)
-            bulb = LIFXLight(
-                self._liffylights, ipaddr, name, power, hue, sat, bri, kel)
-            self._devices.append(bulb)
-            self._add_devices_callback([bulb])
+        self.entities = {}
+        self.hass = hass
+        self.async_add_devices = async_add_devices
+
+    def register(self, device):
```

Comment:
Please mark these classes with `@callback` since they are being called from inside the event loop. This will help readers of this code understand what is going on.